src/components/trie/utils.ts

```diff
@@ -22,46 +33,26 @@ export function trieToTreeUI(
     const branch = root.asBranchNode();
     const leftHash = branch.getLeft();
     const rightHash = branch.getRight();
-    const left = trieToTreeUI(nodes.get(leftHash), nodes, hideEmpty);
-    const right = trieToTreeUI(nodes.get(rightHash), nodes, hideEmpty);
-    const children = [];
 
-    if (shouldRenderNode(leftHash, hideEmpty)) {
-      children.push({
-        name: leftHash.toString(),
-        children:
-          Array.isArray(left) || left === undefined
-            ? left
-            : nodes.get(leftHash)?.getNodeType() === NodeType.Branch
-              ? left.children
-              : [left],
-      });
-    }
-    if (shouldRenderNode(rightHash, hideEmpty)) {
-      children.push({
-        name: rightHash.toString(),
-        children:
-          Array.isArray(right) || right === undefined
-            ? right
-            : nodes.get(rightHash)?.getNodeType() === NodeType.Branch
-              ? right.children
-              : [right],
-      });
-    }
+    const left = trieToTreeUI(nodes.get(leftHash), leftHash, nodes, hideEmpty);
+    const right = trieToTreeUI(nodes.get(rightHash), rightHash, nodes, hideEmpty);
 
     return {
-      name: "Root",
-      children,
+      name: hash.toString(),
+      children: [
+        shouldRenderNode(leftHash, hideEmpty) ? left : null,
+        shouldRenderNode(rightHash, hideEmpty) ? right : null,
+      ].filter((x) => x) as TreeNode[],
     };
   }
 
   const leaf = root.asLeafNode();
   const valueLength = leaf.getValueLength();
 
-  // return { name: `Leaf('${leaf.getKey().toString()}',${value})` };
   return {
     name: leaf.getKey().toString(),
```

Comment:
```suggestion
    name: hash.toString(),
```

Because it's displayed as HASH in the tooltip, we should put hash there.